src/ui/mixins/staleness-mixin.js

```diff
@@ -20,52 +20,192 @@
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
 
+import { isProxy, toRaw } from 'vue';
+
 import StalenessUtils from '@/utils/staleness';
 
 export default {
   data() {
     return {
-      isStale: false
+      staleObjects: [],
+      stalenessSubscription: {},
+      compositionObjectMap: new Map(),
+      setupClockChanged: false
     };
   },
-  beforeUnmount() {
-    this.triggerUnsubscribeFromStaleness();
+  computed: {
+    isStale() {
+      return this.staleObjects.length !== 0;
+    }
   },
   methods: {
-    subscribeToStaleness(domainObject, callback) {
-      if (!this.stalenessUtils) {
-        this.stalenessUtils = new StalenessUtils(this.openmct, domainObject);
+    getSubscriptionId(domainObject) {
+      return (
+        domainObject?.customKey || this.openmct?.objects.makeKeyString(domainObject.identifier)
+      );
+    },
+    setupClockChangedEvent(callback) {
+      this.setupClockChanged = true;
+      this.compositionIteratorCallback = this.compositionIterator(callback);
+      this.openmct.time.on('clockChanged', this.compositionIteratorCallback);
+    },
+    addToCompositionMap(id, domainObject) {
+      if (!this.compositionObjectMap.get(id)) {
+        this.compositionObjectMap.set(id, domainObject);
+      }
+    },
+    compositionIterator(callback) {
+      return () => {
+        this.staleObjects = [];
+        for (const [, object] of this.compositionObjectMap) {
+          let domainObject = object;
+          if (isProxy(domainObject)) {
```

Comment:
We want to keep Vue specific utils (`toRaw`, `isProxy`) in Vue files... I'm guilty of this as well! Do we need this here?

Reply:
I think this is needed here to unwrap the domainObjects and pass them to functions that are expecting them to be unwrapped. Is there a way to access properties on them without converting them from proxies to standard objects? I was getting errors when doing this without the `toRaw` conversion.